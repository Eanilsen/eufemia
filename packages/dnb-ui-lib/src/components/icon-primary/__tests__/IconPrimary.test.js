/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  axeComponent
} from '../../../core/jest/jestSetup'
import Component from '../IconPrimary'

const props = fakeProps(require.resolve('../../icon/IconPrimary'), {
  optional: true
})
props.icon = 'question'
props.size = null

describe('IconPrimary component', () => {
  it('have to match snapshot', () => {
    const Comp = mount(<Component {...props} />)
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('has valid width and height prop', () => {
    const width = 200
    const height = 100
    const Comp = mount(
      <Component {...props} width={width} height={height} />
    )
    const elem = Comp.find('svg')
    expect(elem.exists()).toBe(true)
    expect(elem.props().width).toBe(width)
    expect(elem.props().height).toBe(height)
  })

  it('should validate with ARIA rules', async () => {
    const Comp = mount(<Component {...props} />)
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})
