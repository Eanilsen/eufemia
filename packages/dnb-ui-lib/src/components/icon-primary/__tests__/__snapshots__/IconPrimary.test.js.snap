// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconPrimary component have to match snapshot 1`] = `
<IconPrimary
  alt={null}
  attributes={null}
  border={null}
  className={null}
  color={null}
  height={null}
  icon="question"
  modifier={null}
  size={null}
  title={null}
  width={null}
>
  <span
    alt="question"
    aria-label="question"
    className="dnb-icon dnb-icon--default"
    role="img"
  >
    <question>
      <svg
        fill="none"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          clipRule="evenodd"
          d="M4.35229 5.1004C4.35229 3.04939 5.90168 1.5 7.95269 1.5C10.0037 1.5 11.5531 3.04939 11.5531 5.1004C11.5531 7.15141 10.0037 8.70079 7.95269 8.70079C7.53848 8.70079 7.20269 9.03658 7.20269 9.45079V12.1162C7.20269 12.5304 7.53848 12.8662 7.95269 12.8662C8.36691 12.8662 8.70269 12.5304 8.70269 12.1162V10.1494C11.2089 9.80139 13.0531 7.72428 13.0531 5.1004C13.0531 2.22096 10.8321 0 7.95269 0C5.07325 0 2.85229 2.22096 2.85229 5.1004C2.85229 5.51461 3.18808 5.8504 3.60229 5.8504C4.01651 5.8504 4.35229 5.51461 4.35229 5.1004ZM7.95288 15.9978C8.61088 15.9978 9.04954 15.5591 9.04954 14.9011C9.04954 14.2431 8.61088 13.8044 7.95288 13.8044C7.29488 13.8044 6.85622 14.2431 6.85622 14.9011C6.85622 15.5591 7.29488 15.9978 7.95288 15.9978Z"
          fill="black"
          fillRule="evenodd"
        />
      </svg>
    </question>
  </span>
</IconPrimary>
`;
